src/platform_impl/linux/x11/window.rs

```diff
@@ -120,7 +120,16 @@ impl UnownedWindow {
         pl_attribs: PlatformSpecificWindowBuilderAttributes,
     ) -> Result<UnownedWindow, RootOsError> {
         let xconn = &event_loop.xconn;
-        let root = event_loop.root;
+        // root should be a type ffi::Window and it is finally c_ulong.
+        // cf. https://docs.rs/x11-dl/2.19.1/x11_dl/xlib/type.Window.html
+        let root = if let Some(id) = pl_attribs.parent_id {
+            // XID is defined as 32-bit value in the X11 protocol so
+            // there's no problem about higher bits truncation.
+            // cf. https://www.x.org/docs/XProtocol/proto.pdf
```

Comment:
This comment also feels a bit strange. You can say `WindowId is XID under the hood which doesn't exceed u32, so the conversion is lossless`.

Reply:
fixed